instructionAPI/src/AMDGPU/cdna2/InstructionDecoder-amdgpu-cdna2.h

```diff
@@ -31,7 +31,6 @@
 #include "InstructionDecoderImpl.h"
 #include <iostream>
 #include "Immediate.h"
-#include "dyn_regs.h"
```

Comment:
Now I see that this header is also included in InstructionDecoderImpl.h, so it is safe to remove. However, as the same file is included in InstructionDecoder-aarch64.h [](https://github.com/dyninst/dyninst/blob/a85835269188c24a5c27a50bbd06751c1badc3d7/instructionAPI/src/InstructionDecoder-aarch64.h#L34), won't this cause the same build error? Or this problem only exist for amdgpu cdna2?

Reply:
It won't cause a build error in either place because the macro guards will stop that. I removed it because it wasn't being used. We should remove it from the other files, as well. One of my future tasks is to run IWYU on the whole codebase.

Reply:
I think it was included because some of the functions take MachRegister  as parameter, which is defined in dyn_regs.h

Reply:
The build error I was referring to, is the comment associated with the first commit in this PR

Reply:
Ah, you are right. That makes it worse, not better. Like I said, we can leave that mess until later.

>The build error I was referring to, is the comment associated with the first commit in this PR

Yes. The error there was the absolute path in https://github.com/dyninst/dyninst/blob/a85835269188c24a5c27a50bbd06751c1badc3d7/common/h/dyn_regs.h#L1817 which didn't exist until until #1201.